src/components/CardModal/CardModal.tsx

```diff
@@ -137,9 +122,9 @@ export const CardModal: React.FC<Props> = ({
                 <div className="card-modal__date-close card-modal__date-choose">
                   Expires on
                 </div>
-
+                
                 <DatePicker
-                  selected={startDate}
+                  selected={formatStringToDate(watch('dueDate'))}
```

Comment:
DRY (Dont Repeat Yourself). You have 2 datepickers with selected property, where you formatting date, move to constant. Also do not use watch because it causes to unnecessary rerenders, only useWatch hook. 
const dueDate = useWatch({ name: 'dueDate', control: form.control })
const formattedDate = format...Date(dueDate)